Tests/Factory/ZipkinTracerFactoryTest.php

```diff
@@ -105,4 +127,20 @@ public function testCreateException(): void
             )
         );
     }
+
+    /**
+     * @return array{endpoint_url: string}
+     * @throws \ReflectionException
+     */
+    private function extractReporterOptionsByReflection(\OpenTracing\Tracer $tracer)
```

Comment:
Testing the endpoint_url is pretty ugly, since it is buried deep down in private properties. So I decided to dig into it with Reflection.

Reply:
This is totally reasonable in this case, as the objects are not built to expose this information. I would even not consider requesting such a change upstream, so reflection is fine for us here.